qiskit/qpy/binary_io/circuits.py

```diff
@@ -304,10 +307,19 @@ def _read_instruction(file_obj, circuit, registers, custom_operations, version,
                 params = [len(qargs)]
             elif gate_name in {"BreakLoopOp", "ContinueLoopOp"}:
                 params = [len(qargs), len(cargs)]
-            gate = gate_class(*params)
-        gate.condition = condition
-    if instruction.label_size > 0:
-        gate.label = label
+            if label is not None:
+                if issubclass(gate_class, SingletonGate):
+                    gate = gate_class(*params, label=label)
+                else:
+                    gate = gate_class(*params)
+                    gate.label = label
+            else:
+                gate = gate_class(*params)
+        if condition:
+            if not isinstance(gate, ControlFlowOp):
+                gate = gate.c_if(*condition)
+            else:
+                gate.condition = condition
```

Comment:
lol, all these `if not x: ...; else: ...` always seem so backwards to me, since we need the positive path as well anyway.  No need to change them I guess, though.